scripts/hotfixes/kdump/create-kdump-artifacts.sh

```diff
@@ -0,0 +1,256 @@
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+set -euo pipefail
+
+# Source common dracut parameters.
+. "$(dirname $0)/dracut-lib.sh"
+
+# show the line that we failed and on and exit non-zero
+
+trap 'catch $? $LINENO; cleanup; exit 1' ERR
+
+# if the script is interrupted, run the cleanup function
+trap 'cleanup' INT EXIT
+
+# catch() prints what line the script failed on, runs the cleanup function, and then exits non-zero
+catch() {
+    # Show what line the error occurred on because it can be difficult to detect in a dracut environment
+    echo "CATCH: exit code $1 occurred on line $2 in $(basename "${0}")"
+    cleanup
+    exit 1
+}
+
+
+# cleanup() removes temporary files, puts things back where they belong, etc.
+cleanup() {
+    echo "CLEANUP: cleanup function running ..."
+
+    # Restore the dracut config if it was removed.
+    [ ! -f /etc/dracut.conf.d/05-metal.conf ] && cp -v /run/rootfsbase/etc/dracut.conf.d/05-metal.conf /etc/dracut.conf.d/05-metal.conf
+
+    # Stops re-running this script on reboot
+    # TODO: This script should move to the pipeline, and the service can then be removed (MTL-1830).
+    systemctl disable kdump-cray
+}
+
+
+# check_size() offers a CAUTION message if the initrd is larger then 20MB
+# this is just a soft warning since several factors can influence running out of memory including:
+# crashkernel= parameters, drivers that are loaded, modules that are loaded, etc.
+# so it's more of a your mileage may vary message
+check_size() {
+
+    local initrd="$1"
+    local max=20000000 # kdump initrds larger than 20M may run into issues with memory
+    
+    if [[ "$(stat --format=%s $initrd)" -ge "$max" ]]; then
+        echo >&2 "CAUTION: initrd might be too large ($(stat --format=%s $initrd)) and may OOM if used"
+    else
+        echo "initrd size is $(stat --format=%s $initrd) bytes"
+    fi
+}
+
+
+# FIXME: Remove this function, see notes on each block of code contained for prerequites for removal.
+function update_fstab {
+
+    local initrd="$1"
+    local sqfs_label="${2:-'SQFSRAID'}"
+    local live_dir="${3:-'LiveOS'}"
+
+    echo "Unpacking generated initrd [$initrd] to amend fstab ..."
+
+    local temp=/tmp/ktmp
+
+    [ -d $temp ] && rm -rf $temp
+    mkdir -p $temp
+    pushd $temp || exit 1
+    /usr/lib/dracut/skipcpio ${initrd} | xzcat | cpio -id
+    
+    # This hack removes the automated entry for mnt0, we have been able to resolve where it came from.
```

Comment:
The rest of the line seems superfluous given the following comment line.

```suggestion
    # This hack removes the automated entry for mnt0.
```